### ISSUE TITLE
1.17 fabric version crashes in 1.17.0

### ISSUE THREAD

**Issue by lutravurr** [exaskye/Ears#47]
<details>
---- Minecraft Crash Report ----
// Who set us up the TNT?

Time: 11/27/21, 10:57 PM
Description: Rendering entity in world

java.lang.ArithmeticException: / by zero
	at com.unascribed.ears.common.EarsRenderer.render(EarsRenderer.java:306)
	at com.unascribed.ears.common.EarsCommon.render(EarsCommon.java:131)
	at com.unascribed.ears.common.render.IndirectEarsRenderDelegate.render(IndirectEarsRenderDelegate.java:35)
	at com.unascribed.ears.common.render.IndirectEarsRenderDelegate.render(IndirectEarsRenderDelegate.java:23)
	at com.unascribed.ears.EarsFeatureRenderer.render(EarsFeatureRenderer.java:51)
	at com.unascribed.ears.EarsFeatureRenderer.method_4199(EarsFeatureRenderer.java:37)
	at net.minecraft.class_922.method_4054(class_922.java:203)
	at net.minecraft.class_1007.method_4215(class_1007.java:64)
	at net.minecraft.class_1007.method_3936(class_1007.java:42)
	at net.minecraft.class_898.method_3954(class_898.java:178)
	at net.minecraft.class_490.method_29977(class_490.java:154)
	at com.mojang.blaze3d.systems.RenderSystem.runAsFancy(RenderSystem.java:865)
	at net.minecraft.class_490.method_2486(class_490.java:154)
	at net.minecraft.class_481.method_2389(class_481.java:778)
	at net.minecraft.class_465.method_25394(class_465.java:101)
	at net.minecraft.class_485.method_25394(class_485.java:43)
	at net.minecraft.class_481.method_25394(class_481.java:689)
	at net.minecraft.class_757.method_3192(class_757.java:1279)
	at net.minecraft.class_310.method_1523(class_310.java:1114)
	at net.minecraft.class_310.method_1514(class_310.java:730)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at com.unascribed.ears.common.EarsRenderer.render(EarsRenderer.java:306)
	at com.unascribed.ears.common.EarsCommon.render(EarsCommon.java:131)
	at com.unascribed.ears.common.render.IndirectEarsRenderDelegate.render(IndirectEarsRenderDelegate.java:35)
	at com.unascribed.ears.common.render.IndirectEarsRenderDelegate.render(IndirectEarsRenderDelegate.java:23)
	at com.unascribed.ears.EarsFeatureRenderer.render(EarsFeatureRenderer.java:51)
	at com.unascribed.ears.EarsFeatureRenderer.method_4199(EarsFeatureRenderer.java:37)
	at net.minecraft.class_922.method_4054(class_922.java:203)
	at net.minecraft.class_1007.method_4215(class_1007.java:64)
	at net.minecraft.class_1007.method_3936(class_1007.java:42)

-- Entity being rendered --
Details:
	Entity Type: minecraft:player (net.minecraft.class_746)
	Entity ID: 181
	Entity Name: Kiwnik
	Entity's Exact location: -256.86, 75.00, -499.44
	Entity's Block location: World: (-257,75,-500), Section: (at 15,11,12 in -17,4,-32; chunk contains blocks -272,0,-512 to -257,255,-497), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Entity's Momentum: 0.00, -0.08, 0.00
	Entity's Passengers: []
	Entity's Vehicle: null

-- Renderer details --
Details:
	Assigned renderer: net.minecraft.class_1007@ed346e
	Location: 0.00,0.00,0.00 - World: (0,0,0), Section: (at 0,0,0 in 0,0,0; chunk contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Rotation: 0.0
	Delta: 1.0
Stacktrace:
	at net.minecraft.class_898.method_3954(class_898.java:178)
	at net.minecraft.class_490.method_29977(class_490.java:154)
	at com.mojang.blaze3d.systems.RenderSystem.runAsFancy(RenderSystem.java:865)
	at net.minecraft.class_490.method_2486(class_490.java:154)
	at net.minecraft.class_481.method_2389(class_481.java:778)
	at net.minecraft.class_465.method_25394(class_465.java:101)
	at net.minecraft.class_485.method_25394(class_485.java:43)
	at net.minecraft.class_481.method_25394(class_481.java:689)

-- Screen render details --
Details:
	Screen name: net.minecraft.class_481
	Mouse location: Scaled: (400, 245). Absolute: (1202.000000, 735.000000)
	Screen size: Scaled: (640, 339). Absolute: (1920, 1017). Scale factor of 3.000000

-- Affected level --
Details:
	All players: 1 total; [class_746['Kiwnik'/181, l='ClientLevel', x=-256.86, y=75.00, z=-499.44]]
	Chunk stats: 2025, 1553
	Level dimension: minecraft:overworld
	Level spawn location: World: (-64,86,-256), Section: (at 0,6,0 in -4,5,-16; chunk contains blocks -64,0,-256 to -49,255,-241), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Level time: 12910323 game time, 717733 day time
	Server brand: fabric
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:476)
	at net.minecraft.class_310.method_1587(class_310.java:2399)
	at net.minecraft.class_310.method_1514(class_310.java:749)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:234)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:153)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: Default, Fabric Mods, Rainy Day (1_14).zip, Mizunos 16 Craft JE_1.17-1.0.zip

-- System Details --
Details:
	Minecraft Version: 1.17
	Minecraft Version ID: 1.17
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 16.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 896410640 bytes (854 MiB) / 2147483648 bytes (2048 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 12
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 5 2600X Six-Core Processor           
	Identifier: AuthenticAMD Family 23 Model 8 Stepping 2
	Microarchitecture: Zen+
	Frequency (GHz): 3.59
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: NVIDIA GeForce GT 1030
	Graphics card #0 vendor: NVIDIA (0x10de)
	Graphics card #0 VRAM (MB): 2048.00
	Graphics card #0 deviceId: 0x1d01
	Graphics card #0 versionInfo: DriverVersion=27.21.14.5671
	Memory slot #0 capacity (MB): 4096.00
	Memory slot #0 clockSpeed (GHz): 2.67
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 4096.00
	Memory slot #1 clockSpeed (GHz): 2.67
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 15040.75
	Virtual memory used (MB): 12405.05
	Swap memory total (MB): 6912.00
	Swap memory used (MB): 807.91
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		ears: Ears 1.4.1
		fabric: Fabric API 0.36.0+1.17
		fabric-api-base: Fabric API Base 0.3.0+a02b44633d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.2.0+2b5c62d03d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.11+c345aea83d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+a02b44633d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.1+a02b44633d
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+92519afa3d
		fabric-containers-v0: Fabric Containers (v0) 0.1.12+a02b44633d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.2+a02b44633d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.5+be9da3103d
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.10+a02b44633d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.1.0+a02b44633d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.9+a722d8c03d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+92519afa3d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.6+a02b44633d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.4+a02b44633d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.10+b7ab61213d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+a02b44633d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.2+36b77c3e3d
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.4+a02b44633d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.4+a02b44633d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+92519afa3d
		fabric-models-v0: Fabric Models (v0) 0.3.0+a02b44633d
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.11+b7ab61213d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.11+a02b44633d
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+92519afa3d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.10.9+b7ab61213d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.3+a02b44633d
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+a02b44633d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.10+e2961fee3d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.3+676f40fa3d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.8+a02b44633d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.0+a02b44633d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+a02b44633d
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+a02b44633d
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+92519afa3d
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.0+a02b44633d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.7+b7ab61213d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.4+155f865c3d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.8+a02b44633d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.11+e36969833d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.4+a02b44633d
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+a02b44633d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.12+b7ab61213d
		fabricloader: Fabric Loader 0.11.6
		java: OpenJDK 64-Bit Server VM 16
		minecraft: Minecraft 1.17
		mm: Manningham Mills 2.2
		optifabric: OptiFabric 1.11.1
		xaerominimap: Xaero's Minimap 21.11.5
	Launched Version: fabric-loader-0.11.6-1.17
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: GeForce GT 1030/PCIe/SSE2 GL version 4.6.0 NVIDIA 456.71, NVIDIA Corporation
	Window size: 1920x1017
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Integrated Server (map_client.txt)
	Graphics mode: fancy
	Resource Packs: vanilla, Fabric Mods, file/Rainy Day (1_14).zip (incompatible), file/Mizunos 16 Craft JE_1.17-1.0.zip
	Current Language: English (Canada)
	CPU: 12x AMD Ryzen 5 2600X Six-Core Processor 
	Player Count: 1 / 8; [class_3222['Kiwnik'/181, l='ServerLevel[The Fishtank (Tmk)]', x=-256.86, y=75.00, z=-499.44]]
	Data Packs: vanilla, Fabric Mods
	OptiFine Version: OptiFine_1.17_HD_U_G9_pre21
	OptiFine Build: 20210612-020007
	Render Distance Chunks: 20
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.6.0 NVIDIA 456.71
	OpenGlRenderer: GeForce GT 1030/PCIe/SSE2
	OpenGlVendor: NVIDIA Corporation
	CpuCount: 12

-- OptiFabric --
Details:
	OptiFine jar designed for: 1.17
	OptiFine jar version: OptiFine_1.17_HD_U_G9_pre21
	OptiFine jar status: Valid OptiFine installer
	OptiFine remapped jar: C:/Users/pisce/AppData/Roaming/.minecraft/.optifine/OptiFine_1.17_HD_U_G9_pre21/Optifine-mapped.jar
	OptiFabric error: <None>
</details>	
	
	
	
	~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	
	
	other mods:
	- optifine
	- optifabric
	- fabric api (of course)
	- xaeros minimap
	
	crashes both in solo world and servers

**Comment by Cynosphere** [comment]
Fixed in 1.4.2. For the time being until its fully released, you can use the pre-release of it: https://github.com/unascribed/Ears/releases/tag/v1.4.2-pre2

**Comment by lutravurr** [comment]
@Cynosphere no dice :( just took longer to crash this time

**Comment by exaskye** [comment]
If you're still crashing, it's a different error. Post your crash report. This issue was indeed fixed in 1.4.2.

**Comment by lutravurr** [comment]
this ones 1.16 but it gave me the same error message lmk if you want this in a separate thread o:

<details>
---- Minecraft Crash Report ----

Time: 11/28/21 12:04 AM
Description: Rendering entity in world

java.lang.NullPointerException: Rendering entity in world
	at com.unascribed.ears.common.EarsCommon.shouldSuppressElytra(EarsCommon.java:95)
	at net.minecraft.class_979.handler$zle000$render(class_979.java:523)
	at net.minecraft.class_979.method_17161(class_979.java)
	at net.minecraft.class_979.method_4199(class_979.java:22)
	at net.minecraft.class_922.method_4054(class_922.java:204)
	at net.minecraft.class_1007.method_4215(class_1007.java:65)
	at net.minecraft.class_1007.method_3936(class_1007.java:42)
	at net.minecraft.class_898.method_3954(class_898.java:311)
	at net.minecraft.class_761.method_22977(class_761.java:2120)
	at net.minecraft.class_761.method_22710(class_761.java:1742)
	at net.minecraft.class_757.method_3188(class_757.java:1022)
	at net.minecraft.class_757.method_3192(class_757.java:693)
	at net.minecraft.class_310.method_1523(class_310.java:1048)
	at net.minecraft.class_310.method_1514(class_310.java:681)
	at net.minecraft.client.main.Main.main(Main.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at com.unascribed.ears.common.EarsCommon.shouldSuppressElytra(EarsCommon.java:95)
	at net.minecraft.class_979.handler$zle000$render(class_979.java:523)
	at net.minecraft.class_979.method_17161(class_979.java)
	at net.minecraft.class_979.method_4199(class_979.java:22)
	at net.minecraft.class_922.method_4054(class_922.java:204)
	at net.minecraft.class_1007.method_4215(class_1007.java:65)
	at net.minecraft.class_1007.method_3936(class_1007.java:42)

-- Entity being rendered --
Details:
	Entity Type: minecraft:player (net.minecraft.class_745)
	Entity ID: 2407130
	Entity Name: chigglechen
	Entity's Exact location: -18.65, 70.03, 0.56
	Entity's Block location: World: (-19,70,0), Chunk: (at 13,4,0 in -2,0; contains blocks -32,0,0 to -17,255,15), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Entity's Momentum: 0.00, 0.00, 0.00
	Entity's Passengers: []
	Entity's Vehicle: ~~ERROR~~ NullPointerException: null

-- Renderer details --
Details:
	Assigned renderer: net.minecraft.class_1007@77a7362c
	Location: 2.60,-1.59,-5.37 - World: (2,-2,-6), Chunk: (at 2,-1,10 in 0,-1; contains blocks 0,0,-16 to 15,255,-1), Region: (0,-1; contains chunks 0,-32 to 31,-1, blocks 0,0,-512 to 511,255,-1)
	Rotation: -99.50252
	Delta: 0.68012595
Stacktrace:
	at net.minecraft.class_898.method_3954(class_898.java:311)
	at net.minecraft.class_761.method_22977(class_761.java:2120)
	at net.minecraft.class_761.method_22710(class_761.java:1742)
	at net.minecraft.class_757.method_3188(class_757.java:1022)

-- Affected level --
Details:
	All players: 30 total; [class_746['Kiwnik'/2406930, l='ClientLevel', x=-21.34, y=70.00, z=5.92], class_745['robinjoelsson'/2171624, l='ClientLevel', x=-24.00, y=70.03, z=0.50], class_745['deadshotnoob_yt'/2404243, l='ClientLevel', x=-64.22, y=94.66, z=-9.44], class_745['Boomslinger'/2405212, l='ClientLevel', x=-50.51, y=102.37, z=-2.75], class_745['eyalmal'/2405591, l='ClientLevel', x=-20.53, y=69.00, z=-1.56], class_745['dsjbabykimo1'/2356148, l='ClientLevel', x=-7.84, y=67.00, z=1.88], class_745['LastNight007'/2401254, l='ClientLevel', x=-4.39, y=67.00, z=-6.74], class_745['bradzo_'/2403749, l='ClientLevel', x=-10.86, y=67.00, z=0.01], class_745['Zyphalopagus'/2404240, l='ClientLevel', x=-12.72, y=67.00, z=0.47], class_745['Vibgyor177'/2404446, l='ClientLevel', x=-5.70, y=67.03, z=-9.55], class_745['fo9oasukah'/58, l='ClientLevel', x=-8.50, y=67.00, z=-4.50], class_745['iTzBixaedPvP'/2397347, l='ClientLevel', x=-2.59, y=67.00, z=-8.88], class_745['farmerskyblock'/2398301, l='ClientLevel', x=-27.78, y=73.55, z=9.15], class_745['GothLunch'/2406068, l='ClientLevel', x=-8.03, y=72.28, z=-2.75], class_745['2qwc8z7m9g'/9, l='ClientLevel', x=3.50, y=68.00, z=-2.50], class_745['budjytt7ae'/11, l='ClientLevel', x=-0.50, y=68.00, z=-8.50], class_745['0zv8n4u0r2'/12, l='ClientLevel', x=1.50, y=68.00, z=-5.50], class_745['h131g1v8v5'/10, l='ClientLevel', x=3.50, y=68.00, z=3.50], class_745['0o0t98ospy'/62, l='ClientLevel', x=13.50, y=69.00, z=17.50], class_745['qZachary'/2403849, l='ClientLevel', x=-4.44, y=69.00, z=-21.50], class_745['Megamindbrain'/2403944, l='ClientLevel', x=46.90, y=96.01, z=-21.37], class_745['1l3v1pgdzf'/1, l='ClientLevel', x=3.50, y=67.00, z=-14.50], class_745['lwanz02t67'/2, l='ClientLevel', x=10.50, y=67.00, z=0.50], class_745['055h76qp8y'/3, l='ClientLevel', x=1.50, y=68.00, z=6.50], class_745['6h7760872u'/5, l='ClientLevel', x=-0.50, y=68.00, z=9.50], class_745['6h7760872u'/7, l='ClientLevel', x=-17.50, y=70.00, z=7.50], class_745['Voiidless'/2406933, l='ClientLevel', x=-7.97, y=70.03, z=-7.69], class_745['TheDaVinCiCoDe1'/2406936, l='ClientLevel', x=-11.88, y=67.00, z=-3.50], class_745['edivorce'/2405306, l='ClientLevel', x=-41.09, y=83.00, z=-61.53], class_745['chigglechen'/2407130, l='ClientLevel', x=-18.65, y=70.03, z=0.56]]
	Chunk stats: Client Chunk Cache: 5041, 169
	Level dimension: minecraft:overworld
	Level spawn location: World: (-24,70,0), Chunk: (at 8,4,0 in -2,0; contains blocks -32,0,0 to -17,255,15), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Level time: 227161231 game time, 389000 day time
	Server brand: vanilla
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:617)
	at net.minecraft.class_310.method_1587(class_310.java:2239)
	at net.minecraft.class_310.method_1514(class_310.java:699)
	at net.minecraft.client.main.Main.main(Main.java:215)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.fabricmc.loader.game.MinecraftGameProvider.launch(MinecraftGameProvider.java:226)
	at net.fabricmc.loader.launch.knot.Knot.launch(Knot.java:146)
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28)

-- System Details --
Details:
	Minecraft Version: 1.16.5
	Minecraft Version ID: 1.16.5
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_51, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 1071928864 bytes (1022 MB) / 1845493760 bytes (1760 MB) up to 2147483648 bytes (2048 MB)
	CPUs: 12
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		artifice: Artifice 0.14.6+1.16.5
		cardinal-components: Cardinal Components API 2.8.3
		cardinal-components-base: Cardinal Components API (base) 2.8.3
		cardinal-components-block: Cardinal Components API (blocks) 2.8.3
		cardinal-components-chunk: Cardinal Components API (chunks) 2.8.3
		cardinal-components-entity: Cardinal Components API (entities) 2.8.3
		cardinal-components-item: Cardinal Components API (items) 2.8.3
		cardinal-components-level: Cardinal Components API (world saves) 2.8.3
		cardinal-components-scoreboard: Cardinal Components API (scoreboard) 2.8.3
		cardinal-components-util: Cardinal Components API (utilities) 2.8.3
		cardinal-components-world: Cardinal Components API (worlds) 2.8.3
		cloth-basic-math: Cloth Basic Math 0.5.1
		cloth-config2: Cloth Config v4 4.11.26
		cosmetic-armor: Cosmetic Armor 1.0.0
		curios: Curios API 0.0.13-1.16.5
		ears: Ears 1.4.2-pre2
		enva: Environmental Armor 1.3.1
		extraorigins: Extra Origins 1.16.5-11
		fabric: Fabric API 0.34.6+1.16
		fabric-api-base: Fabric API Base 0.3.0+c88702897d
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.0.0+dc716ea17d
		fabric-biome-api-v1: Fabric Biome API (v1) 3.1.10+3661eaa17d
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.5+ca58154a7d
		fabric-command-api-v1: Fabric Command API (v1) 1.1.1+351679a77d
		fabric-commands-v0: Fabric Commands (v0) 0.2.2+ca58154a7d
		fabric-containers-v0: Fabric Containers (v0) 0.1.11+9354966b7d
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.2.1+ca58154a7d
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.3+ca58154a7d
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.6+61336ba97d
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.1.0+c88702897d
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.2+ca58154a7d
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.1+ca58154a7d
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.6+ca58154a7d
		fabric-item-api-v1: Fabric Item API (v1) 1.2.1+ca58154a7d
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.2.7+2868a2287d
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.4+9354966b7d
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.1+ca58154a7d
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.2.1+ca58154a7d
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.2+ca58154a7d
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.3+ca58154a7d
		fabric-models-v0: Fabric Models (v0) 0.3.0+bc7a746f7d
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.3+e3c9d0627d
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.8+ca58154a7d
		fabric-networking-v0: Fabric Networking (v0) 0.3.2+ca58154a7d
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.9.5+c88702897d
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.2+ca58154a7d
		fabric-particles-v1: Fabric Particles (v1) 0.2.4+ca58154a7d
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.4+ca58154a7d
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.2+2868a2287d
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.4+ca58154a7d
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 2.3.0+2868a2287d
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.5+ca58154a7d
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.1.13+ca58154a7d
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.2+ca58154a7d
		fabric-rendering-v1: Fabric Rendering (v1) 1.6.0+2868a2287d
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.6+2868a2287d
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.0+c045166c7d
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.5+9354966b7d
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.4+ca58154a7d
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.1.1+ca58154a7d
		fabric-textures-v0: Fabric Textures (v0) 1.0.6+ca58154a7d
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.2.7+8183d3957d
		fabricloader: Fabric Loader 0.11.3
		fallflyinglib: FallFlyingLib 1.1.0
		java: Java HotSpot(TM) 64-Bit Server VM 8
		minecraft: Minecraft 1.16.5
		mm: Manningham Mills 2.2
		moborigins: Mob Origins 1.4.0
		optifabric: OptiFabric 1.9.6
		origins: Origins 0.7.3
		origins-classes: Origins: Classes 1.1.1
		originsplus: Origins Plus 0.0.2
		pehkui: Pehkui 2.2.0+21w20a
		reach-entity-attributes: Reach Entity Attributes 1.1.1
		satin: Satin 1.5.1
		slimeorigin: Slimeorigin 3.0.0
		transparent_cosmetics: Transparent Cosmetics 1.2.0
		xaerominimap: Xaero's Minimap 21.9.0.1
	Launched Version: fabric-loader-0.11.3-1.16.5
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: GeForce GT 1030/PCIe/SSE2 GL version 4.6.0 NVIDIA 456.71, NVIDIA Corporation
	GL Caps: Using framebuffer using OpenGL 3.0
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Resource Packs: minecraft:transparent_cosmetics, vanilla, Fabric Mods, file/Rainy Day (1_14).zip (incompatible), file/Mizunos 16 Craft JE_1.16.4-1.0.zip
	Current Language: English (Canada)
	CPU: 12x AMD Ryzen 5 2600X Six-Core Processor 
	OptiFine Version: OptiFine_1.16.5_HD_U_G8
	OptiFine Build: 20210515-161946
	Render Distance Chunks: 20
	Mipmaps: 4
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: null
	OpenGlVersion: 4.6.0 NVIDIA 456.71
	OpenGlRenderer: GeForce GT 1030/PCIe/SSE2
	OpenGlVendor: NVIDIA Corporation
	CpuCount: 12

-- OptiFabric --
Details:
	OptiFine jar designed for: 1.16.5
	OptiFine jar version: OptiFine_1.16.5_HD_U_G8
	OptiFine jar status: Valid OptiFine installer
	OptiFine remapped jar: C:\Users\pisce\AppData\Roaming\.minecraft\.optifine\OptiFine_1.16.5_HD_U_G8\Optifine-mapped.jar
	OptiFabric error: <None>
</details>
